src/main/groovy/life/qbic/portal/infrastructure/services/ServiceFinder.groovy

```diff
@@ -0,0 +1,33 @@
+package life.qbic.portal.infrastructure.services
+
+import groovy.util.logging.Log4j2
+import life.qbic.portal.utils.ConfigurationManagerFactory
+import life.qbic.services.ConsulServiceFactory
+import life.qbic.services.Service
+import life.qbic.services.ServiceConnector
+import life.qbic.services.ServiceType
+import life.qbic.services.connectors.ConsulConnector
+
+@Log4j2
+class ServiceFinder {
+
+    static List<Service> findWorkflowTrackingServices(URL serviceRegistryUrl) throws NullPointerException {
+        ServiceConnector connector = new ConsulConnector()
+        def serviceList = []
+
+        connector.withCloseable {
+            ConsulServiceFactory factory = new ConsulServiceFactory(it)
+            serviceList.addAll(factory.getServicesOfType(ServiceType.WORKFLOW_TRACKING))
+        }
+
+        if (serviceList.isEmpty()) {
```

Comment:
This is boiler code. Just return the list. If it is empty, it is up to the implementation logic to decide what needs to be done. 

Like in the main class, print an error on the console if no service is found, otherwise continue with the application.

Reply:
Basically, this class is not needed, as I do not see its benefit for reusability.

Reply:
In this case this class has the purpose to separate the infrastructure and domain logic. The readability that is gained by having this in the `infrastructure.services` package justifies its existence in my opinion. Cramping everything in the messy Initialisation classes (DependencyManager and PortletEntryPointUI) would not fit in with the responsibilities of the respective classes.